### DISCUSSION TITLE
trying to setup my own server

### DISCUSSION THREAD

**Discussion by 0000matteo0000** [shadowsocks/shadowsocks-rust#538] [General]
I am using these two scripts to set up my server, the server one is run on the server, the client one locally passing the URI as parameter, but I am failing to access internet through the proxy, any suggestions?
The scripts: https://gitlab.com/0000matteo0000/shadowsocks-config/-/tree/master
Even running the server locally and then using a modified URI IP of 127.0.0.1 does not help

**Comment by zonyitoo** [comment]
What error did you see on both these processes?

**Reply by 0000matteo0000** [reply]
The client does not fail, but the browser gives me Unable to connect and tries to load indefinitely, when i try to connect to google.com the clients logs:
`INFO  [1183677:140678050723136] [sslocal] shadowsocks 1.11.0
INFO  [1183677:140678050723136] [shadowsocks_service::local::socks::server] shadowsocks socks TCP listening on 127.0.0.1:1080
INFO  [1183677:140678050723136] [shadowsocks_service::local::socks::server::socks5::udprelay] shadowsocks socks5 UDP listening on 127.0.0.1:1080
DEBUG [1183677:140678050715392] [shadowsocks_service::local::socks::server::socks5::tcprelay] CONNECT google.com:443
DEBUG [1183677:140678050715392] [shadowsocks_service::local::socks::server::socks5::tcprelay] CONNECT google.com:443
DEBUG [1183677:140678040209152] [shadowsocks_service::local::socks::server::socks5::tcprelay] CONNECT google.com:80
DEBUG [1183677:140678050715392] [shadowsocks_service::local::socks::server::socks5::tcprelay] CONNECT google.com:80`
and nothing else

**Reply by zonyitoo** [reply]
What did you see on the server side?

**Reply by 0000matteo0000** [reply]
without the v2ray-plugin and running the server with -v, using "journalctl -e -u shadowsocks.service" i only get
```
May 26 19:28:55 this systemd[1]: Started shadowsocks-rust Server Service.
May 26 19:28:55 this ssserver[15341]: 2021-05-26T19:28:55.245410360+02:00 INFO  [15341:140228247872704] [ssserver] shadowsocks 1.11.0
May 26 19:28:55 this ssserver[15341]: 2021-05-26T19:28:55.246004801+02:00 INFO  [15341:140228247872704] [shadowsocks_service::server::tcprelay] shadowsocks tcp server listening on 0.0.0.0:8388, inbound address 0.0.0.0:8388
May 26 19:28:55 this ssserver[15341]: 2021-05-26T19:28:55.246098078+02:00 INFO  [15341:140228247872704] [shadowsocks_service::server::udprelay] shadowsocks udp server listening on 0.0.0.0:8388
````
and nothing else even when i see connections from the ssclient log

**Reply by 0000matteo0000** [reply]
ok apparently i used a bad ip in the uri not having ports mapped to my local machine, sorry to bother you, you can delete this thread or leave it if you want, i will update the config script shortly.
any reason why the client does not fail or throw error messages if the server connection fails?

**Reply by 0000matteo0000** [reply]
I managed to get the server running and i have updated the scripts but:
"ipv6_first": true  set on both server and client config breaks:
on server: 
```
ERROR tcp tunnel 127.0.0.1:<random_port> -> <random_site>:443 connect failed, error: Network is unreachable (os error 101)`
on client: `Secure Connection Failed PR_END_OF_FILE_ERROR on browser
```

**Reply by zonyitoo** [reply]
You should use \``` for logs to keep the line format.

> any reason why the client does not fail or throw error messages if the server connection fails?

Well, you can set `timeout` in the client's `server` configuration to let it fail if connect too long.

> "ipv6_first": true set on both server and client config breaks

It means that your server and client doesn't have IPv6 network supported. `ipv6_first` is for telling the DNS resolver to use IPv6 addresses first before IPv4 addresses. If you are running on an IPv4 only network and trying to `connect` to IPv6 addresses will result in `Network is unreachable` socket errors.

**Reply by Klaaktu** [reply]
@zonyitoo Hi sorry, some quick questions: Should `ipv6_first` be enabled for server with IPv6 network? Does it increase loading time for website without an IPv6 address? If set to `false` is IPv4 address always preferred?

**Reply by zonyitoo** [reply]
If IPv4 works well, you should leave it alone.

It shouldn't increase loading time, because it always queries IPv4 (`A`) and IPv6 (`AAAA`) simutaneously.

> If set to false is IPv4 address always preferred?

Yes.

**Reply by 0000matteo0000** [reply]
Ok, thanks for the support!

**Reply by x-ji** [reply]
It's funny that I saw this same `error: Network is unreachable (os error 101)` with an IPv6 address, even though I didn't have `ipv6_first` enabled. Turns out that my Firefox has been open for a while, and apparently it tries to connect to the target website via its IPv6 address directly. Using a fresh Safari session worked, while all the time I thought there was something wrong with shadowsocks itself.

**Reply by database64128** [reply]
> Using a fresh Safari session worked

You didn't have to switch browsers. [Happy Eyeballs v2](https://www.rfc-editor.org/rfc/rfc8305) is implemented by all major browsers to automatically select the most optimal IP version. Attempts of opening connections of IPv6 targets won't affect your browsing experience.

**Reply by x-ji** [reply]
Nice to know. It's not clear why using Firefox to visit the target website didn't succeed and resulted in the aforementioned errors at ssserver's side. I guess that shouldn't be the norm.
